mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt

```diff
@@ -0,0 +1,48 @@
+package com.mineinabyss.enchants.enchantments
+
+import com.mineinabyss.enchants.CustomEnchants.Magnetism
+import com.mineinabyss.mineinabyss.core.mineInAbyss
+import com.okkero.skedule.schedule
+import org.bukkit.entity.Item
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.block.BlockBreakEvent
+
+
+class MagnetismListener : Listener {
+    @EventHandler
+    fun BlockBreakEvent.onBlockBreak() {
+
+        val item = player.inventory.itemInMainHand
+
+        if (item.containsEnchantment(Magnetism)){
+
+            val loc = block.location
+
+            mineInAbyss.schedule {
+                waitFor(1)
+
+                val entities = player.world.getNearbyEntities(loc, 4.0, 4.0, 4.0)
+
+                for (entitie in entities) {
+                    if (entitie is Item) {
+                        val item: Item = entitie
+                        val fullInv = player.inventory.firstEmpty() == -1
+                        if (!fullInv){
+                            player.inventory.addItem(item.itemStack)
```

Comment:
I believe addItem retuns any leftover items, you can update the item on ground to have the correctly adjusted stack size depending on this

Reply:
i changed the code for this a bit but im not sure about how i could directly change the stack size on the ground